### PR TITLE
Cumulus 1452

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Use full volume size for docker storage

Addresses [CUMULUS-1452](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1452)

## Changes

* Update the EC2 scripts in terraform to use the volume for docker storage
* Change the default storage driver to devicemapper, overlay2 gives you the base size of the size left on disk

## Testing

* Deploy EC2
* SSH into instance
* Run `sudo vgs`
* Run `sudo lvs`
* Run `docker info | grep "Data Space"`

## PR Checklist

- [X] Update CHANGELOG
- [ ] Unit tests
- [X] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: tf-modules/cumulus/variables.tf [hunk 1]
```diff
@@ -170,7 +170,7 @@ variable "ecs_docker_hub_config" {
 
 variable "ecs_docker_storage_driver" {
   type    = string
-  default = "overlay2"
```
Comment (nemreid) [inline tf-modules/cumulus/variables.tf]: Shouldn't you reproduce this in `packages/deployment/app/config.yml` as well?

Comment (laurenfrederick) [reply tf-modules/cumulus/variables.tf]: Updated

File: tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl [hunk 2]
```diff
@@ -23,13 +23,19 @@ Resources:
...
           Content-Type: text/cloud-boothook; charset="us-ascii"
 
+%{ if docker_storage_driver == "devicemapper" ~}
+          vgcreate docker /dev/xvdcz
+
+          lvcreate -n docker-data -L${docker_volume_create_size}G docker
+
           sed -i '/^\s*DOCKER_STORAGE_OPTIONS=/d' /etc/sysconfig/docker-storage
-          echo 'DOCKER_STORAGE_OPTIONS="--storage-driver ${docker_storage_driver}"' >> /etc/sysconfig/docker-storage
+          echo 'DOCKER_STORAGE_OPTIONS="--storage-driver devicemapper --storage-opt dm.datadev=/dev/docker/docker-data"' >> /etc/sysconfig/docker-storage
```
Comment (nemreid) [inline tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl]: I don't think this --storage-opt flag is present in `app/cloudformation.template/yml`, you may need to update that as well.

Comment (nemreid) [reply tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl]: And check for any additional discrepancies between that script and this one.

Comment (laurenfrederick) [reply tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl]: @nemreid it's not super straightforward in kes and there are workarounds anyway, so opted just to really fix this in TF

Comment (laurenfrederick) [reply tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl]: The issue is you can't allocate for the volume_size, you need to do volume_size + 1, which is super easy in TF and not really worth the trouble in kes

Comment (nemreid) [reply tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl]: OK, makes sense. Thanks.

**Pull request merged**